Amplify/Categories/DataStore/Model/Temporal/Date+Operation.swift

```diff
@@ -0,0 +1,64 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Foundation
+
+public enum DateUnit {
+
+    case days(_ value: Int)
+    case weeks(_ value: Int)
+    case months(_ value: Int)
+    case years(_ value: Int)
+
+    public static let oneDay: DateUnit = .days(1)
+    public static let oneWeek: DateUnit = .weeks(1)
+    public static let oneMonth: DateUnit = .months(1)
+    public static let oneYear: DateUnit = .years(1)
+
+    public var calendarComponent: Calendar.Component {
+        switch self {
+        case .days, .weeks:
+            return .day
+        case .months:
+            return .month
+        case .years:
+            return .year
+        }
+    }
+
+    public var value: Int {
+        switch self {
+        case .days(let value),
+             .months(let value),
+             .years(let value):
+            return value
+        case .weeks(let value):
+            return value * 7
```

Comment:
The `value` property is used in conjunction with `calendarComponent`. When the enum is `.week` the `calendarComponent` is `.day` since there's no `.week` component. Therefore the value needs to be multiplied by 7: `.weeks(2)` is actually `.days(14)` under the hood.

Reply:
I see... hmmm...  The following might be more readable:
```
case .weeks(let weeks):
    let numDaysinWeeks = weeks * 7
    return numDaysinWeeks
```
or we could just leave it alone.  up to you.. thanks for the explanation!